File: data/EXP01/python/2452/2452.output_1.py
# @lc app=leetcode id=2452 lang=python3
class Solution:
    def twoEditWords(self, queries: List[str], dictionary: List[str]) -> List[str]:
        def is_valid_edit(word1: str, word2: str) -> bool:
            if word1 == word2:
                return True
            elif len(word1) != len(word2):
                return False
            else:
                diff_count = 0
                for i in range(len(word1)):
                    if word1[i] != word2[i]:
                        diff_count += 1
                    if diff_count > 2:
                        return False
                return True
        
        valid_words = []
        for query in queries:
            for d in dictionary:
                if is_valid_edit(query, d):
                    valid_words.append(query)
                    break
        return valid_words

File: data/EXP01/golang_no_system_var/2452/2452.output_1.go
func twoEditWords(queries []string, dictionary []string) []string {
    result := []string{}
    
    for _, query := range queries {
        if contains(dictionary, query) {
            result = append(result, query)
        } else {
            for _, word := range dictionary {
                if isTransformable(query, word) {
                    result = append(result, query)
                    break
                }
            }
        }
    }
    
    return result
}

func contains(arr []string, word string) bool {
    for _, w := range arr {
        if w == word {
            return true
        }
    }
    return false
}

func isTransformable(query, word string) bool {
    diffCount := 0
    for i := 0; i < len(query); i++ {
        if query[i] != word[i] {
            diffCount++
            if diffCount > 2 {
                return false
            }
        }
    }
    return true
}
